File: code/src/models/surveys.ts
import { Model } from "sequelize";

interface ISurveysStatus extends Model {
  loaded_at: string;
  status: string;
  count: number;
}

type TSurveysRequest = {
  createdAt: string;
  origin: string;
  limit: number;
  offset: number;
};

enum SurveyTypeRequest {
  STATUS = "STATUS",
  ORIGINS = "ORIGINS",
}

interface ISurveysRequest {
  createdAt?: { start?: string; end?: string };
  origin?: string;
  limit?: number;
  offset?: number;
  type?: SurveyTypeRequest;
}

type StatusData = {
  Aberto?: number;
  Pendente?: number;
  Valido?: number;
  Invalido?: number;
  Visualizou?: number;
  Incompleto?: number;
};

type ISurveysStatusResponse = {
  date: string;
  status: StatusData;
};

export {
  ISurveysStatus,
  TSurveysRequest,
  ISurveysRequest,
  ISurveysStatusResponse,
};

File: code/surveys-status-1.2/src/app/models/surveys.py
from datetime import datetime

from pydantic import BaseModel, Field


class CreatedAt(BaseModel):
    start: datetime | None = None
    end: datetime | None = None


class SurveysStatusRequest(BaseModel):
    created_at: CreatedAt | None = Field(alias="createdAt", default=None)
    limit: int | None = Field(ge=0, default=None)
    offset: int | None = Field(ge=0, default=None)


class Status(BaseModel):
    aberto: int | None = Field(default=None)
    pendente: int | None = Field(default=None)
    valido: int | None = Field(default=None)
    invalido: int | None = Field(default=None)
    visualizou: int | None = Field(default=None)
    incompleto: int | None = Field(default=None)


class SurveysStatusResponse(BaseModel):
    date: datetime
    total: int
    status: Status
